### DISCUSSION TITLE
Feature: Record Device Audio for screen recordings

### DISCUSSION THREAD

**Discussion by salad612** [Bnyro/RecordYou#232] [Ideas]
Option to instead of or with microphone, record device audio

**Comment by xmha97** [comment]
Good idea 💡
Like DroidRec app

**Comment by Wjxfi** [comment]
Look at the code https://www.f-droid.org/packages/com.yakovlevegor.DroidRec/
he did it

**Comment by Bnyro** [comment]
I know, but their code doesn't really have a structure and the code for recording screen and device audio is more than 1k lines...

**Comment by NeeRaj-2401** [comment]
Does device/system audio term includes on going call sounds too
I mean will it includ recording "on going" system i.e. call audios recording too?

**Comment by salad612** [comment]
> Does device/system audio term includes on going call sounds too I mean will it includ recording "on going" system i.e. call audios recording too?

Haven't tested but according to this maybe?
![audio](https://user-images.githubusercontent.com/121117961/221286610-6599a55d-49de-449b-939e-fa3d4afbb27f.png)

**Comment by salad612** [comment]
> I know, but their code doesn't really have a structure and the code for recording screen and device audio is more than 1k lines...

maybe you could ask the maintainer. I'm sure he'd be happy to point out what the key part is

**Comment by NeeRaj-2401** [comment]
> Haven't tested but according to this maybe? ![audio](https://user-images.githubusercontent.com/121117961/221286610-6599a55d-49de-449b-939e-fa3d4afbb27f.png)

@salad612 is that ur system application or any 3rd party ?

**Comment by salad612** [comment]
> > Haven't tested but according to this maybe? ![audio](https://user-images.githubusercontent.com/121117961/221286610-6599a55d-49de-449b-939e-fa3d4afbb27f.png)
> 
> @salad612 is that ur system application or any 3rd party ?

System app, I think it's on Android 11+. Droidrec might work too though

**Comment by ElsAr4e** [comment]
Maybe you could also add the option to capture Unprocessed (Raw) audio even if you are using Android 8,9 (non-rooted devices):

https://www.youtube.com/watch?v=h52RLrs3ySw

**Comment by yusef388** [comment]
From Android 10+ Google have added the "AudioPlaybackCapture API" to capture Internal app's audio like games and streaming. please consider to adding it to this great app. other recording app on store have this feature but with so many ads pretty annoying. 

Have a look at this API:
[https://developer.android.com/guide/topics/media/playback-capture](url)

**Comment by Red6785** [comment]
Please.. This is all this app needs to fully replace DroidRec for me, which seems to have vanished off the face of GitHub.
